### PR TITLE
feat: add a maintainer guide for jenkins x

### PR DESCRIPTION
**Pull request by ankitm123**
# Description

We dont have a maintainer guide, this is an attempt at adding one.
Hopefully, we will make it better with time.

Fixes # (issue)

# Checklist:

- [ ] I have mentioned the appropriate type(scope), as referenced [here](https://jenkins-x.io/community/code/#the-commit-message) in the commit message and PR title for the semantic checks to pass.
- [ ] I have signed off the commit, as per instructions mentioned [here](https://jenkins-x.io/community/code/#how-to-sign-your-commits).
- [ ] Any dependent changes have already been merged.

### CODE REVIEW

File: content/en/community/maintainer_guide/infrastructure.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- GKE kubernetes version 1.21
+- Spot instances to keep costs low
+  - n1-standard-2 with boot size 100 GB
+- Cluster autoscaler scales the nodes between 1 and 5.
+- We use terraform, an infrastructure as Code tool to manage the cluster using the [google-jx](https://github.com/jenkins-x/terraform-google-jx) module.
+
+### jenkins-x-bdd
+
+This project is only used for bdd testing.
+Test clusters are created here when a pull request is opened in the [jx3-versions](https://github.com/jenkins-x/jx3-versions) repository.
```
Comment (tomhobson) [inline content/en/community/maintainer_guide/infrastructure.md:30]: should we mention the gc things here?

**Review by tomhobson — CHANGES_REQUESTED** [review]

File: content/en/community/maintainer_guide/mentorship/_index.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+title: Mentorship opportunities
+linktitle: Mentorship
+description: Everything a Jenkins X maintainer should know about
+weight: 100
+type: docs
+no_list: true
+---
+
+As a maintainer, one of the primary tasks is to help expand the adoption of Jenkins X and on board new contributors.
+
```
Comment (babadofar) [inline content/en/community/maintainer_guide/mentorship/_index.md:11]: "and to onboard new contributors"

File: content/en/community/maintainer_guide/infrastructure.md (hunk 1 above)
Comment (ankitm123) [reply content/en/community/maintainer_guide/infrastructure.md:30]: Done

File: content/en/community/maintainer_guide/mentorship/_index.md (hunk 2 above)
Comment (ankitm123) [reply content/en/community/maintainer_guide/mentorship/_index.md:11]: Done.

**Review by tomhobson — APPROVED** [review]

**Pull request merged**